pkg/proxy/ipvs/proxier.go

```diff
@@ -1656,15 +1662,7 @@ func (proxier *Proxier) syncProxyRules() {
 	legacyBindAddrs := proxier.getLegacyBindAddr(activeBindAddrs, currentBindAddrs)
 
 	// Clean up legacy IPVS services and unbind addresses
-	appliedSvcs, err := proxier.ipvs.GetVirtualServers()
-	if err == nil {
-		for _, appliedSvc := range appliedSvcs {
-			currentIPVSServices[appliedSvc.String()] = appliedSvc
-		}
-	} else {
-		klog.ErrorS(err, "Failed to get ipvs service")
-	}
-	proxier.cleanLegacyService(activeIPVSServices, currentIPVSServices, legacyBindAddrs)
+	proxier.cleanLegacyService(activeIPVSServices, legacyBindAddrs)
```

Comment:
I do wonder if we should call `refreshCurrentIPVSServices` one more time before calling `cleanLegacyService`. Otherwise, currentIPVServices is stale. But maybe that is okay because we are only deleting at this point and `activeIPVSServices` would be accurate.